arch/sim/src/sim/win/up_hosttime.c

```diff
@@ -0,0 +1,122 @@
+/****************************************************************************
+ * arch/sim/src/sim/win/up_hosttime.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <windows.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* 100ns difference between Windows and UNIX timebase. */
+
+#define FACTOR (0x19db1ded53e8000LL)
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: host_gettime
+ ****************************************************************************/
+
+uint64_t host_gettime(bool rtc)
+{
+  static LONGLONG start;
+  LARGE_INTEGER now;
+
+  GetSystemTimeAsFileTime((LPFILETIME)&now);
+
+  now.QuadPart -= FACTOR;
+  now.QuadPart *= 100;
+
+  if (rtc)
+    {
+      return now.QuadPart;
+    }
+
+  if (start == 0)
+    {
+      start = now.QuadPart;
+    }
+
+  return now.QuadPart - start;
+}
+
+/****************************************************************************
+ * Name: host_sleep
+ ****************************************************************************/
+
+void host_sleep(uint64_t nsec)
+{
+  LARGE_INTEGER due;
+  HANDLE timer;
+
+  /* Convert to 100 nanosecond interval,
+   * negative value indicates relative time
+   */
+
+  due.QuadPart = -(10 * ((nsec + 999) / 1000));
+
+  timer = CreateWaitableTimer(NULL, TRUE, NULL);
```

Comment:
call Sleep instead

Reply:
Precision of Sleep() does not match with host api